src/nvim/getchar.c

```diff
@@ -1644,7 +1644,10 @@ int vgetc(void)
   may_garbage_collect = false;
 
   // Execute Lua on_key callbacks.
+  int save_got_int = got_int;
+  got_int = false;
   nlua_execute_on_key(c);
+  got_int = save_got_int;
```

Comment:
can this stuff live in `nlua_execute_on_key` ?

Reply:
Yes.